src/nifs.rs

```diff
@@ -310,8 +310,17 @@ mod tests {
     };
 
     // create a shape object
+    let rows = num_cons;
+    let cols = num_vars + num_io + 1;
```

Comment:
Here the number of cols is set to num_vars + num_io + 1. Whereas earlier, it seems off by one. Is there a reason?

Reply:
This is perhaps a stylistic departure from the idiomatic Nova style: num_io here is a variable that does not contain the extra dimension for one. But note this is self-coherent as the call to R1CSShape is being passed `num_io`.

I've added 27aa13e to correct.

Reply:
> This is perhaps a stylistic departure from the idiomatic Nova style: num_io

I'm not sure it was a departure. In Nova, Scalar::one is not counted as part of public IO, so num_vars + num_io + 1 was quite consistent.